tasks/Module 3/task_3.py

```diff
@@ -0,0 +1,36 @@
+def generate_numbers(N:int, M:int, prefix=None):
```

Comment:
You are missing whitespace between `N` and `int`. Could you scan your code with pylint and fix all relevant styling errors? 

I really enjoy the fact that you are playing with typing. Could you also add type for prefix/other functions in your codebase? Eg. tasks from module 3.

The naming convention could be applied here, how about changing `N` and `M` to lowercase?

From what I understood `prefix` is a `list`. It's a little bit confusing as on line 12 in `gen_bin` function `prefix` is a string.

If you need to add type to `list` you need to import it from `typing` module
`from typing import List`
and use it like
`prefix: List[str]` 

You can read more about typing in https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html